dotCMS/src/main/java/com/dotmarketing/util/ResourceCollectorUtil.java

```diff
@@ -111,8 +109,11 @@ private static Collection<String> removeVersionRange(String packages){
         }
         
         return finalSet;
-        
-        
+    }
+
+    public static boolean isValidPackage (final String packageName) {
+
+        return packageName.length() > 0 && !".".equals(packageName); // more?
```

Comment:
In its current usage I understand this will not be passed a null packageName,  but as you have made this public it seems the intention is that it is part of the Util and therefore is intended to be used from other code, it would be good to be explicit and prevent a null pointer exception proactively.  This could be just returning false in the case of null if we expect and don't care if this happens, or could add Objects.requireNonNull(packageName, "packageName should never be null");

Reply:
I have started the defensive from here, but at the end move the effort to another part. However I thing would be a good idea remove public scope to protected and add the NPE, thx